forum_leaderboard/forum_leaderboard.py

```diff
@@ -0,0 +1,111 @@
+"""
+Forum Leaderboard XBlock
+
+Shows the top threads for a given discussion ID by vote.
+"""
+from django.template import Template, Context
+
+import pkg_resources
+
+from xblock.core import XBlock
+from xblock.fields import Scope, Integer, String
+from xblock.fragment import Fragment
+
+import lms.lib.comment_client as cc
+
+
+@XBlock.needs('discussion')
+class ForumLeaderboardXBlock(XBlock):
+    """
+    TO-DO: document what your XBlock does.
+    """
+    display_name = String(
+        default="Forum Leaderboard", scope=Scope.settings,
+        help="Display name for this block."
+    )
+    # Studio view not yet written. To test, set the default here,
+    # or manually set in Mongo.
+    discussion_id = String(
+        default="", scope=Scope.settings,
+        help="The ID of the inline discussion to tally leading threads for.",
+    )
+    count = Integer(
+        default=10, scope=Scope.settings,
+        help="How many threads to display."
+    )
+
+    def resource_string(self, path):
+        """
+        Handy helper for getting resources from our kit.
+        """
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    def create_fragment(self, html, context=None, css=None, javascript=None, initialize=None):
+        """
+        Create an XBlock, given an HTML resource string and an optional context, list of CSS
+        resource strings, list of JavaScript resource strings, and initialization function name.
+        """
+        html = Template(self.resource_string(html))
+        context = context or {}
+        css = css or []
+        javascript = javascript or []
+        frag = Fragment(html.render(Context(context)))
+        for sheet in css:
+            frag.add_css(self.resource_string(sheet))
+        for script in javascript:
+            frag.add_javascript(self.resource_string(script))
+        if initialize:
+            frag.initialize_js(initialize)
+        return frag
+
+    def student_view(self, context=None):
+        """ Renders student view for LMS and Studio """
+        if hasattr(self, 'xmodule_runtime') and getattr(self.xmodule_runtime, 'is_author_mode', False):
+            fragment = self._student_view_studio()
+        else:
+            fragment = self._student_view_lms()
```

Comment:
@bradenmacdonald Just tried this. Turns out that the Studio does not have a configuration that allows it to contact the forum by default. It will go to the wrong port number and get a connection refused error. So the separate views will have to stay, but for a different reason.